gossip/src/crds_value.rs

```diff
@@ -494,80 +499,125 @@ impl Sanitize for NodeInstance {
 pub struct RestartLastVotedForkSlots {
     pub from: Pubkey,
     pub wallclock: u64,
-    pub slots: Vec<CompressedSlots>,
+    pub compressed_slots: Vec<u8>,
+    pub uncompressed_bytes: u64,
+    pub last_voted_slot: Slot,
     pub last_voted_hash: Hash,
     pub shred_version: u16,
 }
 
 impl Sanitize for RestartLastVotedForkSlots {
     fn sanitize(&self) -> std::result::Result<(), SanitizeError> {
-        if self.slots.is_empty() {
+        if self.last_voted_slot == 0 {
             return Err(SanitizeError::InvalidValue);
         }
-        self.slots.sanitize()?;
         self.last_voted_hash.sanitize()
     }
 }
 
 impl RestartLastVotedForkSlots {
-    pub fn new(from: Pubkey, now: u64, last_voted_hash: Hash, shred_version: u16) -> Self {
-        Self {
+    pub fn new(
+        from: Pubkey,
+        now: u64,
+        last_voted_fork: &mut [Slot],
+        last_voted_hash: Hash,
+        shred_version: u16,
+    ) -> Result<Self, String> {
+        let last_voted_slot;
+        let mut compressed_slots;
+        let uncompressed_bytes;
+        if last_voted_fork.is_empty() {
+            return Err("Last voted slot must be specified".to_string());
+        } else {
+            last_voted_fork.sort_by_key(|a| Reverse(*a));
+            last_voted_slot = last_voted_fork[0];
```

Comment:
`[0]` can panic. There is `last_voted_fork.is_empty()` above , but the better style is to remove that and instead change this line to:
```rust
let Some(last_voted_slot) = last_voted_fork.get(0).copied() else {
   return Err(...);
}
```